### PR TITLE
Add shared workflows for NodeJS and generic Docker workflow

### PR DESCRIPTION
**Pull request by 91dave**
- Refactor the docker build & push to ECR out of the `serverless.yml` workflow (new `docker.yml` file)
- New `node-tests.yml` workflow
- Refactor documentation
- List future work and improvements needed

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -16,20 +16,26 @@ on:
...
-      STAGE:       # Serverless Stage - Also used for DK_TAG
-        required: true
+      DK_TAG:       # Docker Tag
+        required: false
+        type: string
+        default: 'latest'
+      DK_CONTEXT:
+        required: false
         type: string
+        default: '.'
```
Comment (robert396) [inline .github/workflows/docker.yml:26]: Based on how this is actually being used, this is more of a working directory context and less of a direct docker context.

**Review by robert396 — APPROVED** [review]
overall this is looking pretty good.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (91dave) [reply .github/workflows/docker.yml:26]: Agree - will change this and re-request review. Will use `WORKDIR` as parameter if that sounds sensible.

Comment (robert396) [reply .github/workflows/docker.yml:26]: makes sense to me.

**Review by robert396 — APPROVED** [review]

**Pull request merged**